### ISSUE TITLE
`updater=nothing` means pass `o` or the initial belief/`missing`

### ISSUE THREAD

**Issue by zsunberg** [JuliaPOMDP/POMDPs.jl#189]
I was working with @sharonzhou a few weeks ago on a project, and one thing she wanted to do was use a `generate_a_given_o` policy (if I understand correctly).

Part of this is passing the `o` straight to the policy without doing a belief update. This has been possible for some time with `PreviousObservationUpdater`, but that is rather verbose. Moreover, `PreviousObservationUpdater` is rather messy because there are different ways of initializing the observation - one implementation uses Nullable; another is not type stable; etc.

I think perhaps we should upgrade this to something more first-class like saying that setting the updater to `nothing` means that the observation should be passed directly to the policy. That is
```julia
update(::Void, b, a, o) = o
```

When Julia v1.0 drops, it will be efficient to use small unions of types (e.g. `Union{MyObs, Missing}` or `Union{MyObs, MyBelief}`), so the case when an observation is not available isn't so awkward like it is now.

One question to settle is whether we should have
```julia
initialize_belief(::Void, d) = d
```
i.e. the argument to the policy on the first step will be the initial state distribution, or
```julia
initialize_belief(::Void, d) = missing
```
i.e. the argument to the policy on the first step will be `missing`

**Comment by zsunberg** [comment]
Do people think this is a good idea?

**Comment by MaximeBouton** [comment]
Making `PreviousObservationUpdater` more simple and easier to use is a good idea.  I found it very useful when working with memoryless policies. 
Right now, the usage for simulations is a bit complicated, I think this also has to do with how `simulate` is implemented. Here is an example:
```
up = FastPreviousObservationUpdater{MyObs}()
b0 = initialize_belief(up, o0)
simulate(hr, pomdp, policy, up, b0, s0)
```

Why setting the updater to `nothing` ? Isn't it clearer with a custom type?

**Comment by zsunberg** [comment]
Yeah, I think most of the annoying complexity was due to type stability concerns - when I made it, I didn't know if it was best to use Nullables, require the updater to be primed with an initial observation, or cause type-instability.

After 1.0, since small unions of types are efficient, the solution is obvious - return an observation when available, missing when it's not.

The only question is what to call it. `PreviousObservationUpdater` is kinda awkward, and it would break backwards compatibility (maybe that doesn't matter). I think `nothing` makes as much sense as any name. We could also have `nothing` do the same thing as `PreviousObservationUpdater` or some other thing we decide to call it.
